src/ir_Bosch.cpp

```diff
@@ -25,37 +27,298 @@ const uint16_t kBoschZeroSpace = 571;
+    case kBosch144Heat: return stdAc::opmode_t::kHeat;
+    case kBosch144Dry: return stdAc::opmode_t::kDry;
+    case kBosch144Fan: return stdAc::opmode_t::kFan;
+    default: return stdAc::opmode_t::kAuto;
+  }
+}
+
+/// Convert a native fan speed into its stdAc equivalent.
+/// @param[in] speed The native setting to be converted.
+/// @return The stdAc equivalent of the native setting.
+stdAc::fanspeed_t IRBosch144AC::toCommonFanSpeed( const uint16_t speed )
+{
+  switch( speed )
+  {
+    case kBosch144Fan100: return stdAc::fanspeed_t::kMax;
+    case kBosch144Fan80: return stdAc::fanspeed_t::kHigh;
+    case kBosch144Fan60: return stdAc::fanspeed_t::kMedium;
+    case kBosch144Fan40: return stdAc::fanspeed_t::kLow;
+    case kBosch144Fan20: return stdAc::fanspeed_t::kMin;
+    default: return stdAc::fanspeed_t::kAuto;
+  }
+}
+
+/// Convert the current internal state into its stdAc::state_t equivalent.
+/// @return The stdAc equivalent of the native settings.
+stdAc::state_t IRBosch144AC::toCommon( void ) const 
+{
+  stdAc::state_t result{};
+  result.protocol = decode_type_t::BOSCH144;
+  result.power = getPower();
+  result.mode = toCommonMode( ( _.ModeS1 << 1 ) + _.ModeS3 );
+  result.celsius = true;
+  result.degrees = getTemp();
+  result.fanspeed = toCommonFanSpeed( ( _.FanS1 << 6 ) + _.FanS3 );
+  result.quiet = getQuiet();
+  // Not supported.
+  result.model = -1;
+  result.turbo = false;
+  result.swingv = stdAc::swingv_t::kOff;
+  result.swingh = stdAc::swingh_t::kOff;
+  result.light = false;
+  result.filter = false;
+  result.econo = false;
+  result.clean = false;
+  result.beep = false;
+  result.clock = -1;
+  result.sleep = -1;
+  return result;
+}
+
+/// Convert the current internal state into a human readable string.
+/// @return A human readable string.
+String IRBosch144AC::toString( void ) const 
+{
+  uint8_t mode = ( _.ModeS1 << 1 )+_.ModeS3;
+  uint16_t fan = ( _.FanS1 << 6 ) + _.FanS3;
+  String result = "";
+  result.reserve( 70 );  // Reserve some heap for the string to reduce fragging.
+  result += addBoolToString( getPower(), kPowerStr, false );
+  result += addModeToString( mode, kBosch144Auto, kBosch144Cool,
+                            kBosch144Heat, kBosch144Dry, kBosch144Fan);
+  result += addFanToString( fan>>1, kBosch144Fan100>>1, kBosch144Fan20>>1,
+                           kBosch144FanAuto>>1, kBosch144FanAuto>>1,
+                           kBosch144Fan60>>1);
+  result += addTempToString( getTemp() );
+  return result;
+}
+
+void IRBosch144AC::setInvertBytes()
+{
+  _.InnvertS1_1 = ~_.raw[ 0 ];
+  _.InnvertS1_2 = ~_.raw[ 2 ];
+  _.InnvertS1_3 = ~_.raw[ 4 ];
+  _.InnvertS2_1 = ~_.raw[ 6 ];
+  _.InnvertS2_2 = ~_.raw[ 8 ];
+  _.InnvertS2_3 = ~_.raw[ 10 ];
+}
+
+void IRBosch144AC::setCheckSumS3()
+{
+  _.ChecksumS3 =  _.raw[ 12 ];
+  _.ChecksumS3 += _.raw[ 13 ];
+  _.ChecksumS3 += _.raw[ 14 ];
+  _.ChecksumS3 += _.raw[ 15 ];
+  _.ChecksumS3 += _.raw[ 16 ];
```

Comment:
You can use `sumBytes()` from:
https://github.com/crankyoldgit/IRremoteESP8266/blob/2bad02785274a54b5208228d80c159e0826f193a/src/IRutils.h#L34-L35
e.g.
```c++
  _.ChecksumS3 =  sumBytes(_.raw + 12, 5);
```
or
```c++
  _.ChecksumS3 =  sumBytes(&(_.raw[12]), 5);
```

Reply:
changed